app/src/main/java/com/xmartlabs/gong/ui/screens/signin/SignInScreen.kt

```diff
@@ -1,54 +1,78 @@
+import androidx.compose.foundation.text.KeyboardActions
+import androidx.compose.foundation.text.KeyboardOptions
 import androidx.compose.material.Button
 import androidx.compose.material.MaterialTheme
 import androidx.compose.material.Scaffold
 import androidx.compose.material.Text
 import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.collectAsState
 import androidx.compose.runtime.getValue
-import androidx.compose.runtime.livedata.observeAsState
+import androidx.compose.runtime.remember
+import androidx.compose.ui.ExperimentalComposeUiApi
 import androidx.compose.ui.Modifier
+import androidx.compose.ui.focus.FocusRequester
+import androidx.compose.ui.focus.focusRequester
 import androidx.compose.ui.platform.LocalContext
-import androidx.compose.ui.platform.LocalLifecycleOwner
+import androidx.compose.ui.platform.LocalSoftwareKeyboardController
+import androidx.compose.ui.text.input.ImeAction
 import androidx.compose.ui.text.style.TextAlign
 import androidx.compose.ui.tooling.preview.Preview
 import androidx.compose.ui.unit.dp
 import androidx.constraintlayout.compose.ConstraintLayout
-import androidx.lifecycle.LifecycleOwner
 import androidx.navigation.NavHostController
 import androidx.navigation.compose.navigate
 import androidx.navigation.compose.popUpTo
 import com.xmartlabs.gong.ui.Screens
-import com.xmartlabs.gong.ui.common.extensions.observeStateResult
 import com.xmartlabs.gong.ui.composables.RoundedCornersPasswordTextField
 import com.xmartlabs.gong.ui.composables.RoundedCornersTextField
 import com.xmartlabs.gong.ui.theme.AppTheme
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.onEach
 import org.koin.androidx.compose.getViewModel
 import java.util.Locale
 
 @Composable
 fun SignInScreen(navController: NavHostController) {
   val viewModel: SignInScreenViewModel = getViewModel()
-  val user by viewModel.userLiveData.observeAsState(initial = "")
-  val password by viewModel.passwordLiveData.observeAsState(initial = "")
-  val lifecycleOwner = LocalLifecycleOwner.current
+  val state by viewModel.state.collectAsState()
   val context = LocalContext.current
+  // We only want the event stream to be attached once
+  // even if there are multiple re-compositions
+  LaunchedEffect(null) {
+    viewModel.oneShotEvents
+        .onEach { event ->
+          when (event) {
+            SignInViewModelEvent.NavigateToDashboard ->
+              navController.navigate(Screens.WELCOME) {
+                popUpTo(Screens.SIGN_IN) { inclusive = true }
+              }
+            is SignInViewModelEvent.SignInError -> {
+              if (event.throwable is SecurityException) {
+                Toast.makeText(
+                    context,
+                    "password or username is wrong, try with userId = 'xmartlabs', password 'xmartlabs'",
+                    Toast.LENGTH_SHORT
+                ).show()
```

Comment:
Maybe have functions that handle this events so this portion of code is more readable?